### PR TITLE
add dual-stack support for GCP

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
Adds dual-stack support for GCP. Considers pod CIDR information passed in command line flags. 

**Which issue(s) this PR fixes** 

Implements part of https://github.com/kubermatic/kubermatic/issues/9350

**Optional Release Note**:

```release-note
add dual-stack support for GCP
```

### CODE REVIEW

File: pkg/cloudprovider/provider/gce/service.go [hunk 1]
```diff
@@ -72,16 +80,66 @@ func (svc *service) networkInterfaces(cfg *config) ([]*compute.NetworkInterface,
...
+
+				ifc.Ipv6AccessConfigs = []*compute.AccessConfig{
+					{
+						Name:        "external-ipv6",
+						NetworkTier: "PREMIUM", // TODO: check if ipv6 is supported in other tiers
+						Type:        "DIRECT_IPV6",
+					},
+				}
+			} else {
+				klog.Infof("dualstack is not supported in %q zone", cfg.zone)
```
Comment (PratikDeoghare) [inline pkg/cloudprovider/provider/gce/service.go:114]: error here

File: hack/run-machine-controller.sh [hunk 2]
```diff
@@ -22,13 +22,5 @@ MC_KUBECONFIG=${MC_KUBECONFIG:-$(dirname $0)/../.kubeconfig}
...
-  -worker-count=50 \
-  -logtostderr \
-  -v=6 \
-  -cluster-dns=172.16.0.10 \
-  -enable-profiling \
-  -metrics-address=0.0.0.0:8080 \
-  -use-osm \
-  -health-probe-address=0.0.0.0:8085 \
-  -node-container-runtime=containerd
+  -kubeconfig=/home/pratik/Downloads/kubeconfig-admin-65mwlchzr2 \
```
Comment (rastislavs) [inline hack/run-machine-controller.sh:25]: Did you forget to undo this change?

Comment (PratikDeoghare) [reply hack/run-machine-controller.sh:25]: dang! yes

File: pkg/cloudprovider/provider/gce/service.go [hunk 3]
```diff
@@ -72,16 +80,66 @@ func (svc *service) networkInterfaces(cfg *config) ([]*compute.NetworkInterface,
...
+		if util.ContainsCIDR(networkConfig.PodCIDRs, util.IPv4) &&
+			util.ContainsCIDR(networkConfig.PodCIDRs, util.IPv6) {
+
+			if isIPv6Suported(cfg.zone) {
+				ifc.StackType = "IPV4_IPV6"
+
+				ifc.Ipv6AccessConfigs = []*compute.AccessConfig{
+					{
+						Name:        "external-ipv6",
+						NetworkTier: "PREMIUM", // TODO: check if ipv6 is supported in other tiers
```
Comment (rastislavs) [inline pkg/cloudprovider/provider/gce/service.go:109]: I think that only premium is supported ATM

File: pkg/cloudprovider/provider/gce/service.go [hunk 4]
```diff
@@ -72,16 +80,66 @@ func (svc *service) networkInterfaces(cfg *config) ([]*compute.NetworkInterface,
...
+
+		} else {
+			klog.Infof("pod cidr %q doesn't specify dual stack cluster", networkConfig.PodCIDRs)
+		}
 	}
 
 	return []*compute.NetworkInterface{ifc}, nil
 }
 
+func isIPv6Suported(zone string) bool {
```
Comment (moadqassem) [inline pkg/cloudprovider/provider/gce/service.go:125]: This check should be added in the webhook in the `Vaildate` method. So that you validate the support for IPv6 before you create the Machine

Comment (moadqassem) [reply pkg/cloudprovider/provider/gce/service.go:125]: Also there is a typo s/isIPv6Suported/isIPv6Supported

**Comment by PratikDeoghare** [comment]
Fixing...

**Comment by PratikDeoghare** [comment]
Replaced by https://github.com/kubermatic/machine-controller/pull/1236
